dcos/cosmospackage.py

```diff
@@ -310,13 +310,14 @@ def experimental_package_add(self, dcos_package):
         """
         request = "add"
         headers = self._request_preferences().get(request).pop(0)
-        headers['X-Dcos-Content-Length'] = str(_file_length(dcos_package))
-        data = _chunk_file(dcos_package)
-        response = self._post(request, params=None, headers=[headers], data=data)
+        with util.open_file(dcos_package, 'r+b') as pkg:
```

Comment:
Use `'rb'`. You don't need to write to the file: https://docs.python.org/2/library/functions.html#open